Changing Codegen section name in Root Signature proposal (#106)
llvm/wg-hlsl: proposals/NNNN-root-signature-dx-containe-dx-container-codegen.md

<!-- {% raw %} -->

# Implementation of Root Signatures in DX Container

* Proposal: [NNNN](NNNN-filename.md)
* Author(s): [Xiang Li](https//github.com/python3kgae), [Damyan
  Pepper](https://github.com/damyanp), [Joao Saffran](https://github.com/joaosaffran)
* Status: **Design In Progress**

<!--
*During the review process, add the following fields as needed:*

* PRs: [#NNNN](https://github.com/microsoft/DirectXShaderCompiler/pull/NNNN)
* Issues:
  [#NNNN](https://github.com/microsoft/DirectXShaderCompiler/issues/NNNN)
  -->

## Introduction

> [Purposal 0002](https://github.com/joaosaffran/wg-hlsl/blob/purposal/root-signatures/proposals/0002-root-signature-in-clang.md?plain=1)

Reviewer: ```suggestion
[Proposal 0002](0002-root-signature-in-clang.md)
```

"Proposal" is spelled incorrectly, but more importantly this links to a version in your fork which isn't the right thing to do.

adds root signature into LLVM IR. The remaining support to generated DXContainer 
in a binary serialized format need to be added into clang.

This change proposes adding:

* Conversion of the metadata representation to the binary serialized format.
* Testing solutions to validate root signature blob generation within DX Container.

## Motivation

> The motivation is the same as the one in [Purposal 0002](https://github.com/joaosaffran/wg-hlsl/blob/purposal/root-signatures/proposals/0002-root-signature-in-clang.md#motivation).

Reviewer: ```suggestion
The motivation is the same as the one in [Proposal 0002](0002-root-signature-in-clang.md#motivation).
```

## Proposed solution

> ### Generating of Root Signatures BLOB

Reviewer: ```suggestion
### Generating of Root Signature blob
```
